src/base/bittorrent/session.cpp

```diff
@@ -2645,6 +2654,62 @@ void Session::setAltGlobalUploadSpeedLimit(int limit)
         configureDeferred();
 }
 
+int Session::altPauseDownloads() const
+{
+    return m_isAltPauseDownloadsEnabled;
+}
+
+void Session::setAltPauseDownloads(bool enabled)
+{
+    m_isAltPauseDownloadsEnabled = enabled;
+}
+
+int Session::altPauseUploads() const
+{
+    return m_isAltPauseUploadsEnabled;
+}
+
+void Session::setAltPauseUploads(bool enabled)
+{
+    m_isAltPauseUploadsEnabled = enabled;
+}
+
+void Session::applyAltPauseDownloads(bool enabled)
+{
+    if (enabled) {
+        foreach (TorrentHandle *const torrent, m_torrents) {
+            if (torrent->isDownloading()) {
+                torrent->pause();
+            }
+        }
+    }
+    else {
+        foreach (TorrentHandle *const torrent, m_torrents) {
+            //torrent is paused and not completed, resume download
+            if (torrent->isPaused() && !torrent->isCompleted())
```

Comment:
```suggestion
            if (torrent->isPaused() && !torrent->isCompleted() && !torrent->isErrored())
```
Or
```suggestion
            if (torrent->isPaused() && !torrent->isCompleted()
                && !torrent->hasError() && !torrent->hasMissingFiles())
```

Not sure if there is any difference between these two.

Reply:
Looks like there are no differences. I will use first suggestion.